felix/dataplane/linux/bpf_route_mgr.go

```diff
@@ -379,11 +385,15 @@ func (m *bpfRouteManager) calculateRoute(cidr ip.CIDR) routes.ValueInterface {
 	case proto.RouteType_REMOTE_TUNNEL:
 		flags |= routes.FlagsRemoteTunneledHost
 		route = m.bpfOps.NewValueWithNextHop(flags, cidr.Addr())
-	case proto.RouteType_LOCAL_HOST:
+	case proto.RouteType_LOCAL_HOST, proto.RouteType_BLACK_HOLE:
 		// It may be a localhost IP that is not assigned to a device like an
 		// k8s ExternalIP. Route resolver knew that it was assigned to our
 		// hostname.
-		flags |= routes.FlagsLocalHost
+		if cgRoute.Type == proto.RouteType_LOCAL_HOST {
+			flags |= routes.FlagsLocalHost
+		} else {
+			flags = routes.FlagBlackHole
+		}
 		fallthrough
```

Comment:
woudn't it be much cleaner to get rid of the fallthrough and either call `route = m.bpfOps.NewValue(flags)` directly with the right flags for each case or take the default branch out and fo `if route == nil && flags != 0 { ... }`